runtime-interface/src/lib.rs

```diff
@@ -171,42 +171,37 @@ pub trait GearRI {
     /// If `protect` argument is true then restrict all accesses to page,
     /// else allows read and write accesses.
     fn mprotect_lazy_pages(wasm_mem_addr: u64, protect: bool) -> Result<(), MprotectError> {
-        let mut prev_page = 0u32;
-        let mut size_in_pages = 0u32;
-        let mut lazy_pages = gear_lazy_pages::get_lazy_pages_numbers();
+        let lazy_pages = gear_lazy_pages::get_lazy_pages_numbers();
+        if lazy_pages.is_empty() {
+            return Ok(());
+        }
 
         if let Some(page) = lazy_pages.last() {
             // This case is impossible in real live, so just returns err and does nothing.
```

Comment:
I know this is not your changes nevertheless maybe edit the text to `... impossible in reality`?

Reply:
I'd remove this part at all 😄 But I agree to rename at least.